src/webOAuthServer.ts

```diff
@@ -399,6 +405,22 @@ export class WebServer extends AsyncCreatable<WebServer.Options> {
     this.redirectStatus.emit('complete');
   }
 
+  /**
+   * Preflight request:
+   *
+   * https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request
+   * https://www.w3.org/TR/2020/SPSD-cors-20200602/#resource-preflight-requests
+   */
+  public handlePreflightRequest(response: http.ServerResponse): void {
+    // We don't validate the origin here because:
+    // 1. The default login URL (login.salesforce.com) will not match after a redirect or if user choose a custom domain in login.
+    // 2. There's no fixed list of auth URLs we could check against.
```

Comment:
point 1 refers to the oauth config object that can be accessed here.

> If the value of the [Origin](https://www.w3.org/TR/2020/SPSD-cors-20200602/#http-origin) header is not a [case-sensitive](https://www.w3.org/TR/2020/SPSD-cors-20200602/#case-sensitive) match for any of the values in [list of origins](https://www.w3.org/TR/2020/SPSD-cors-20200602/#list-of-origins) do not set any additional headers and terminate this set of steps.
Always matching is acceptable since the [list of origins](https://www.w3.org/TR/2020/SPSD-cors-20200602/#list-of-origins) can be unbounded.